### PR TITLE
Created FirestoreDB class; Added Post retrieval and upload

### PR DESCRIPTION
**Pull request by lemohammed**
Changelog:
- Created FirestoreDB clas. Query, Add document, fetch the document, get upload timestamp
- Changed User class to use Firestone statics
- Created Post and User interfaces for Firestore documents
- Changed post class to be compatible with PostDoc interface
- Change PostContainer to render Posts with new modifications
- Added correct types to Comments and Posts
- Created WritePost component and  View
- Created loading component
- Added loading component to Private Routing and Write Post View
- Added Write Post route to App.tsx
- Exported firebase types from firebase module

### CODE REVIEW

**Review by rjb75 — APPROVED** [review]
Great job @lemohammed! 🏆

File: src/components/HomePostView.tsx [hunk 1]
```diff
@@ -1,18 +1,40 @@
...
+        getPosts();
+    }, []);
     return (
         <IonContent>
-            <PostContainer postData={testPost}></PostContainer>
-            <PostContainer postData={testPost}></PostContainer>
-            <PostContainer postData={testPost}></PostContainer>
-            <PostContainer postData={testPost}></PostContainer>
+            {posts
+                ? posts.map((v, k) => {
```
Comment (RatikKapoor) [inline src/components/HomePostView.tsx:29]: 😍

**Review by RatikKapoor — CHANGES_REQUESTED** [review]
@lemohammed Just quickly edit your PR for the changelog and then 🚀 😄

**Review by RatikKapoor — APPROVED** [review]
🚀

**Pull request merged**